devito/dle/backends/yask.py

```diff
@@ -93,27 +258,164 @@ def nary2binary(args, op):
 
         def run(expr):
             if expr.is_Integer:
-                return fac.new_const_number_node(int(expr))
+                return YASK.nfac.new_const_number_node(int(expr))
             elif expr.is_Float:
-                return fac.new_const_number_node(float(expr))
+                return YASK.nfac.new_const_number_node(float(expr))
+            elif expr.is_Symbol:
+                assert expr in self.mapper
+                return self.mapper[expr]
             elif isinstance(expr, Indexed):
                 function = expr.base.function
                 assert function.name in self.grids
                 indices = [int((i.origin if isinstance(i, LoweredDimension) else i) - j)
                            for i, j in zip(expr.indices, function.indices)]
                 return self.grids[function.name].new_relative_grid_point(*indices)
             elif expr.is_Add:
-                return nary2binary(expr.args, fac.new_add_node)
+                return nary2binary(expr.args, YASK.nfac.new_add_node)
             elif expr.is_Mul:
-                return nary2binary(expr.args, fac.new_multiply_node)
+                return nary2binary(expr.args, YASK.nfac.new_multiply_node)
             elif expr.is_Pow:
                 num, den = expr.as_numer_denom()
                 if num == 1:
-                    return fac.new_divide_node(run(num), run(den))
+                    return YASK.nfac.new_divide_node(run(num), run(den))
             elif expr.is_Equality:
-                return fac.new_equation_node(*[run(i) for i in expr.args])
+                if expr.lhs.is_Symbol:
+                    assert expr.lhs not in self.mapper
+                    self.mapper[expr.lhs] = run(expr.rhs)
+                else:
+                    return YASK.nfac.new_equation_node(*[run(i) for i in expr.args])
             else:
                 dle_warning("Missing handler in Devito-YASK translation")
                 raise NotImplementedError
 
         return run(expr)
+
+
+# YASK interface
+
+def init(dimensions, shape, dtype, architecture='hsw', isa='avx2'):
+    """
+    To be called prior to any YASK-related operation.
+
+    A new bootstrap is required wheneven any of the following change: ::
+
+        * YASK version
+        * Target architecture (``architecture`` param)
+        * Floating-point precision (``dtype`` param)
+        * Domain dimensions (``dimensions`` param)
+        * Folding
+        * Grid memory layout scheme
+    """
+    global YASK
+
+    if YASK is not None:
+        return
+
+    dle("Initializing YASK...")
+
+    try:
+        import yask_compiler as yc
+        # YASK compiler factories
+        cfac = yc.yc_factory()
+        nfac = yc.yc_node_factory()
+    except ImportError:
+        _force_exit("Python YASK compiler bindings")
+
+    try:
+        # Set directory for generated code
+        path = os.path.join(os.environ['YASK_HOME'], 'src', 'kernel', 'gen')
```

Comment:
Yes, this is how it will work
When YASK will be ready, probably we'll update the installation instructions to ship a virtual env, with these magic variables automatically set